operations/configuration_management/Target_Ansible_Tasks_for_Image_Customization.md

```diff
@@ -3,29 +3,75 @@
+
+`cray_cfs_image` can be used at the top of the playbook with Ansible's `group_by` module.
+This turns the variable into a host group, and allows Ansible to more efficiently exclude tasks from being run, rather than checking a conditional for every task.
+
+```yaml
+- name: group_by cray_cfs_image
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: Classify hosts by whether they're an image
+      group_by:
+        key: image_{{ cray_cfs_image | default(false) | lower }}
+
+# Image customization
+- name: Image customization
+  hosts: all:&image_true
+  gather_facts: false
+  tasks:
+    ...
+
+# Node personalization
+- name: Node personalization
+  hosts: all:&image_false
+  gather_facts: false
+  tasks:
+    ...
+```
 
-It is also best practice to include a default in Ansible roles for playbook and role portability because CFS injects this variable at runtime. This can be done in the defaults section of the role, or where the variable is called. For example:
+When multiple playbooks are imported from a top level playbook, the `group_by` task can also be moved to the top  level playbook so that it only needs to run once
 
 ```yaml
-when: "{{ cray_cfs_image | default(false) }}"
+- name: group_by cray_cfs_image
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: Classify hosts by whether they're an image
+      group_by:
+        key: image_{{ cray_cfs_image | default(false) | lower }}
+
+- import_playbook: playbook1.yaml
+- import_playbook: playbook2.yaml
 ```
 
-If a default is not provided, any playbooks or roles will not be runnable outside of the CFS Ansible Execution Environment \(AEE\) without the user specifying `cray_cfs_image` in the `vars` files or with the Ansible `extra-vars` options.
+See Ansible's [playbooks best practices](https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#handling-os-and-distro-differences) guide for more information on `group_by`
 
-CFS automatically sets this variable in the `hosts/01-cfs-generated.yaml` file for all sessions. When the session target is image customization, it sets `cray_cfs_image` to `true`; otherwise, it is `false`.
+### Using `cray_cfs_image` with conditionals
+
+When `group_by` cannot be used, `cray_cfs_image` can be set with [Ansible playbook conditionals](https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html)
+to selectively run individual tasks with `when: cray_cfs_image`, or to ignore individual tasks with `when: not cray_cfs_image`.
+
+Conditionals can also be applied to entire roles if desired \(see the external [apply Ansible conditionals to roles](https://docs.ansible.com/ansible/latest/user_guide/playbooks_conditionals.html#applying-when-to-roles-imports-and-includes)\).
+In instances where the same playbook may be run in both modes, it is best practice to include a conditional on all parts of the playbook. This is best done by placing the conditional on an `include_*` statement.
```

Comment:
With this line, I can't tell whether you're promoting the group_by module or the use of include_*. You mention group_by first and include_* second. You do push group_by in the second document.

Reply:
Outdated